Add support for data-* attributes via bindings and on WFLink.

apinstein opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/apinstein/phocoa/57)
<!-- Reviewable:end -->

apinstein commented:
@leftdevel if you can check this out and LMK your thoughts!

leftdevel commented on phocoa/framework/widgets/WFView.php:
```diff
@@ -287,6 +292,35 @@ public static function exposedProperties()
+     */
+    protected function renderDataAttributesToHTML()
+    {
+        if (count($this->dataAttributes) === 0)
```
This seems redundant. The for loop would immediately exit if no attributes, and then we'd return the empty html string.

It could help readability & debugging, but since the rest of the code is barely 2 lines, I find this not that useful.

apinstein replied:
fair enough. I think I wrote this as a guard as the first thing I wrote. I'll fix it.

leftdevel commented on phocoa/framework/widgets/WFView.php:
```diff
@@ -287,6 +292,35 @@ public static function exposedProperties()
+            return '';
+        }
+
+        $html = ' ';
```
Why is the empty space needed? `$html = ' '`.  There's already padding at line 317, at the start and end of quotes.

apinstein commented:
@leftdevel I fixed all noted issues and moved it to reviewable. LMK if they look good!

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/apinstein/phocoa/57#-:-KMWYoqXHSWfXWaUcL9q:-1609479928)*
<!-- Sent from Reviewable.io -->

leftdevel commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/apinstein/phocoa/57#-:-KMWdQDHwhwh20bNBdHf:1417088721)*
<!-- Sent from Reviewable.io -->
